_posts/2015-11-21-happy-trees.md

```diff
@@ -0,0 +1,167 @@
+
+My first attempt resulted in this image (click for 5000x5000 version):
+
+[![Result of first attempt](/imgs/happy-tree/happy-tree-blue-small.png)](/imgs/happy-tree/happy-tree-blue.png)
+
+The first thing you'll notice is that it looks pretty neat.
+
+The second thing you'll notice is that there's actually only one melancoil in
+the 6-digit hexadecimal number set. The innermost black circle is `000000` which
+only happifies to itself, and nothing else will happify to it (sad `000000`).
+The second circle represents `000001`, and all of its runty children. And
+finally the melancoil, comprised of: `00000D`, `0000A9`, `0000B5`, `000092`,
+`000055`, and `000032`.
+
+The final thing you'll notice (or maybe it was the first, since it's really
+obvious) is that it's very blue. Non-blue colors are really only represented as
+leaves on their trees and don't ever really have any children of their own, so
+the blue and black sections take up vastly more space.
+
+This makes sense. The number which should generate the largest happification
+result, `FFFFFF`, only results in `000546`, which is primarily blue. So in effect
+all colors happify to some shade of blue.
+
+This might have been it, technically this is the happification tree and the
+melancoil of 6 digit hexadecimal numbers represented as colors. But it's also
+boring, and I wanted to do better.
+
+## Second attempt
+
+The root of the problem is that the definition of "happification" I used
+resulted in not diverse enough results. I wanted something which would give me
+numbers where any of the digits could be anything. Something more random.
+
+I considered using a hash instead, like md5, but that has its own problems.
+There's no gaurantee that any number would actually reach `000001`, which isn't
+required but it's a nice feature that I wanted. It also would be unlikely that
+there would be any melancoils that weren't absolutely gigantic.
+
+I ended up redefining what it meant to happify a hexadecimal number. Instead of
+adding all the digits up, I first split up the red, green, and blue digits into
+their own numbers, happified those numbers, and finally reassembled the results
+back into a single number. For example:
+
+```
+FF5500
+FF, 55, 00
+F*F + F*F, 5*5 + 5*5, 0*0 + 0*0
+1C2, 32, 00
+C23200
+```
+
+I drop that 1 on the `1C2`, because it has no place in this system. Sorry 1.
+
+Simply replacing that function resulted in this image (click for 5000x5000) version:
+
+[![Result of second attempt](/imgs/happy-tree/happy-tree-colors-small.png)](/imgs/happy-tree/happy-tree-colors.png)
+
+The first thing you notice is that it's so colorful! So that goal was achieved.
+
+The second thing you notice is that there's *significantly* more melancoils.
+Hundreds, even. I'll list out a couple of them here:
+
+```
+00000D 0000A9 0000B5 000092 000055 000032
+000D0D 00A9A9 00B5B5 009292 005555 003232
+0D0D0D A9A9A9 B5B5B5 929292 555555 323232
+0D0D32 A9A90D B5B5A9 9292B5 555592 323255
+...
+```
```

Comment:
I don't understand, is this a list of numbers within a single coil, or a list of numbers with each their own coil?